data/actions/scripts/other/potions.lua

```diff
@@ -27,132 +72,90 @@ function onUse(player, item, fromPosition, target, toPosition, isHotkey)
 	if target == nil or not target:isPlayer() then
 		return true
 	end
-
+	
 	if player:getCondition(CONDITION_EXHAUST_HEAL) then
 		player:sendTextMessage(MESSAGE_STATUS_SMALL, Game.getReturnMessage(RETURNVALUE_YOUAREEXHAUSTED))
 		return true
 	end
 
-	local itemId = item:getId()
-	if itemId == antidotePot then
-		if not antidote:execute(target, numberToVariant(target:getId())) then
-			return false
-		end
-
+	local stats = potions[item:getId()]
+	if stats then
+		if (stats.vocations or stats.minLevel) and not player:getGroup():getAccess() then
+			if not stats.minLevel then
+				stats.minLevel = 1
+			end
+			if not stats.vocations then
+				stats.vocations = {}
+			end
+			
+			local vocationList, mvoc = {}, player:getVocation()
+			while mvoc do
```

Comment:
That's it, If you want the potion for lower vocation just put the lower vocation, and the script realizes the higher vocation.

Because it should also work for custom servers, if we want "Super Mana Rune" for only promoted of higher.